src/client/dfuse/SConscript

```diff
@@ -52,18 +52,28 @@ def build_common(env, files, is_shared):
 def build_client_libs_shared(env, prereqs):
     """build the shared interception library"""
 
+    # Both libioil and libpil4dfs need hook.c. Compile hook.c first.
+    hookenv = env.Clone()
+    if '-fvisibility=hidden' in hookenv['CFLAGS']:
+        hookenv['CFLAGS'].remove('-fvisibility=hidden')
+    hookenv.AppendUnique(CFLAGS=['-fPIC'])
+    hookenv.require('capstone')
+    hook_obj = hookenv.SharedObject(os.path.join('pil4dfs', 'hook.c'), SHOBJPREFIX='s_')
+
     ilenv = env.Clone()
     ilenv.AppendUnique(CFLAGS=['-fPIC'])
     ilenv.AppendUnique(CPPDEFINES=['IOIL_PRELOAD'])
     ilenv.AppendUnique(LIBPATH=[Dir('../dfs')])
     ilenv.AppendUnique(LIBPATH=[Dir('../api')])
     ilenv.AppendUnique(LIBS=['dfs'])
+    ilenv.require('capstone')
     penv = ilenv.Clone()
     penv.AppendUnique(CPPDEFINES=['_FILE_OFFSET_BITS=64'])
 
     il_obj = []
     for src in IOIL_SRC:
         il_obj += ilenv.SharedObject(os.path.join('il', src), SHOBJPREFIX='s_')
```

Comment:
ah, this is why you didn't get a warning because of SHOBJPREFIX.   Regardless, better to build things once so thanks for fixing it.